api/app/Http/Controllers/EntityController.php

```diff
@@ -201,13 +217,13 @@ public function patchMany(Request $request)
      */
     public function deleteOne($id)
     {
-        $this->validateId($id, $this->getKeyName(), $this->validateRequestRule);
+        $this->validateId($id, $this->getModel()->getKeyName(), $this->validateRequestRule);
 
         try {
-            $model = $this->getRepository()->find($id);
-            $this->getRepository()->delete($model);
+            $model = $this->getEntityById($id);
+            $model->delete();
         } catch (ModelNotFoundException $e) {
-            throw $this->notFoundException($this->getKeyName());
+            throw $this->notFoundException($this->getModel()->getKeyName());
```

Comment:
This try-catch block can be eliminated by the find method being extracted to `$this->findEntityOrFail($id)`